packages/node/src/utils/redisCache.ts

```diff
@@ -0,0 +1,77 @@
+import type { CommandArgs as IORedisCommandArgs } from '@opentelemetry/instrumentation-ioredis';
+
+export const GET_COMMANDS = ['get', 'mget'];
+export const SET_COMMANDS = ['set' /* todo: 'setex' */];
+// todo: del, expire
+
+/** Determine cache operation based on redis statement */
+export function getCacheOperation(
+  statement: string,
+): 'cache.get' | 'cache.put' | 'cache.remove' | 'cache.flush' | undefined {
+  const lowercaseStatement = statement.toLowerCase();
+
+  if (GET_COMMANDS.includes(lowercaseStatement)) {
+    return 'cache.get';
+  } else if (SET_COMMANDS.includes(lowercaseStatement)) {
+    return 'cache.put';
+  } else {
+    return undefined;
+  }
+}
+
+function keyHasPrefix(key: string, prefixes: string[]): boolean {
+  return prefixes.some(prefix => key.startsWith(prefix));
+}
+
+/** Safely converts a redis key to a string (comma-separated if there are multiple keys) */
+export function getCacheKeySafely(cmdArgs: IORedisCommandArgs): string {
+  try {
+    if (cmdArgs.length === 0) {
+      return '';
+    }
+
+    const keys: string[] = [];
+
+    cmdArgs.forEach(arg => {
+      if (typeof arg === 'string' || typeof arg === 'number') {
+        keys.push(arg.toString());
+      } else if (Buffer.isBuffer(arg)) {
+        keys.push(arg.toString());
+      } else if (Array.isArray(arg)) {
+        arg.forEach(subArg => {
+          if (typeof subArg === 'string' || typeof subArg === 'number') {
+            keys.push(subArg.toString());
+          } else if (Buffer.isBuffer(subArg)) {
+            keys.push(subArg.toString());
+          }
```

Comment:
We should probably add an else block here adding something like `< unknown >` or so, that you at least see how many things there are? Otherwise you may see e.g.

`cache:x, cache:y` when in reality it may be `cache:x, cache:y, cache:BINARY_THINK`?